src/components/TokenNFTPreview.js

```diff
@@ -1,24 +1,38 @@
 import React, { useState, useEffect } from 'react';
+import TagManager from 'react-gtm-module';
 
 const TokenNFTPreview = (props) => {
   const [token, setToken] = useState(props.token);
 
   useEffect(() => {
     // ensure data binding with props.token re rendering when changed
     setToken(props.token);
+    TagManager.dataLayer({
+      dataLayer: {
+        'event': 'LoadNFTPreview'
```

Comment:
Can we add here the token ID in the event? So we can know which token is being loaded more? Or this event already group by token because of the URL?

Reply:
Yes we can. Just checked if the page url is sent with the event, and it is. However for better use i've decided to add the `token.uid` as well.